src/validation.cpp

```diff
@@ -2379,7 +2429,7 @@ static bool ActivateBestChainStep(CValidationState& state, const CChainParams& c
 
         // Connect new blocks.
         BOOST_REVERSE_FOREACH(CBlockIndex *pindexConnect, vpindexToConnect) {
-            if (!ConnectTip(state, chainparams, pindexConnect, pindexConnect == pindexMostWork ? pblock : std::shared_ptr<const CBlock>(), connectTrace)) {
+            if (!ConnectTip(state, chainparams, pindexConnect, pindexConnect == pindexMostWork ? pblock : std::shared_ptr<const CBlock>(), connectTrace, disconnectpool)) {
```

Comment:
I don't know what the best answer here is.  I could call ReacceptToMemoryPool(), but doing so after we've invoked AbortNode() seems either annoying (trying to lookup UTXOs from disk when we're potentially having disk trouble seems bad), or potentially problematic (is it okay to trigger SyncTransaction() callbacks when our on-disk state is potentially corrupt?). 

One idea: check to see if fShutdownRequested is set, and if so, just wipe the disconnectpool and the mempool so that any observers of the mempool won't see a potentially inconsistent state (and if fShutdownRequested is not set, call ReacceptToMemoryPool() -- this should never happen).  Then the mempool saving won't work in the event of an AbortNode(), but maybe that's okay, we're probably out of disk space anyway.

@sipa Any suggestions on how to handle this?